app/api/groups/get_group_progress.go

```diff
@@ -72,67 +72,80 @@ func (srv *Service) getGroupProgress(w http.ResponseWriter, r *http.Request) ser
 		return service.NoError
 	}
 
-	var dbResult []map[string]interface{}
-	// It still takes about 3 minutes to complete on large data sets
-	service.MustNotBeError(srv.Store.GroupAncestors().
-		Select(`
-			groups_ancestors.idGroupAncestor AS idGroup,
+	users := srv.Store.GroupGroups().
+		Select("child.ID").
+		Joins("JOIN groups AS parent ON parent.ID = groups_groups.idGroupParent AND parent.sType != 'Team'").
+		Joins("JOIN groups AS child ON child.ID = groups_groups.idGroupChild and child.sType = 'UserSelf'").
+		Joins(`
+			JOIN groups_ancestors
+			ON groups_ancestors.idGroupAncestor IN (?) AND
+				groups_ancestors.idGroupChild = parent.ID`, ancestorGroupIDs).
```

Comment:
why `parent.ID` and not `child.ID`?

Reply:
It is because here we check that a user is a member of some non-team group that is a descendant of some ancestor group. (The child is a user group)

Reply:
But if he would be the direct child of one of the `ancestorGroupIDs`, why would he be excluded from this "users" variable?
... auto-answer: ah ... because a group is its own ancestor so he would be not be excluded. Ok then. 🙂

Reply:
But if he would be the direct child of one of the ancestorGroupIDs, why would he be excluded from this "users" variable?
... auto-answer: ah ... because a group is its own ancestor so he would be not be excluded. Ok then. 🙂